tfhe/src/high_level_api/integers/types/compact.rs

```diff
@@ -90,3 +92,284 @@ where
         })
     }
 }
+
+impl<P: IntegerParameter> ParameterSetConformant for GenericCompactInteger<P> {
+    type ParameterSet = PBSParameters;
+    fn is_conformant(&self, pbs_parameters: &PBSParameters) -> bool {
+        let params = CtListConformanceParams {
+            shortint_params: pbs_parameters.to_shortint_conformance_param(),
+            num_blocks_per_integer: P::num_blocks(),
+            num_integers_constraint: ListSizeConstraint {
+                max_group_count: 1,
+                group_size: 1,
+            },
+        };
+        self.list.is_conformant(&params)
+    }
+}
+
+impl<P: IntegerParameter> Named for GenericCompactInteger<P> {
+    const NAME: &'static str = "high_level_api::GenericCompactInteger";
+}
+
+impl<P: IntegerParameter> ParameterSetConformant for GenericCompactIntegerList<P> {
+    type ParameterSet = (PBSParameters, ListSizeConstraint);
+    fn is_conformant(
+        &self,
+        (pbs_parameters, number_integers_constraint): &(PBSParameters, ListSizeConstraint),
+    ) -> bool {
+        let params = CtListConformanceParams {
+            shortint_params: pbs_parameters.to_shortint_conformance_param(),
+            num_blocks_per_integer: P::num_blocks(),
+            num_integers_constraint: *number_integers_constraint,
+        };
+        self.list.is_conformant(&params)
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::conformance::ParameterSetConformant;
+    use crate::core_crypto::prelude::UnsignedInteger;
+    use crate::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
```

Comment:
let's plan adding more tests later down the line using the create_parameterized_test pattern from elsewhere in the crate

Reply:
I don't really see the benefit to hav this test with many parameter sets (except if they increase coverage, for exemple by using a non native modulus)

Reply:
it's just a matter of completeness :)

plus the macro instantiation of tests is really not that bad